dotnet_sdk/lib/dependabot/dotnet_sdk/file_parser.rb

```diff
@@ -0,0 +1,80 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "dependabot/dependency"
+require "dependabot/file_parsers"
+require "dependabot/file_parsers/base"
+require "sorbet-runtime"
+
+module Dependabot
+  module DotnetSdk
+    class FileParser < Dependabot::FileParsers::Base
+      extend T::Sig
+
+      require "dependabot/file_parsers/base/dependency_set"
+
+      sig { override.returns(T::Array[Dependabot::Dependency]) }
+      def parse
+        dependency_set = DependencySet.new
+
+        config_dependency_files.each do |config_dependency_file|
+          dependency = parse_dependency_file(config_dependency_file)
+          dependency_set << dependency if dependency
+        end
+
+        dependency_set.dependencies
+      end
+
+      private
+
+      sig { params(dependency_file: Dependabot::DependencyFile).returns(T.nilable(Dependabot::Dependency)) }
+      def parse_dependency_file(dependency_file)
+        return unless dependency_file.content
+
+        begin
+          contents = JSON.parse(T.must(dependency_file.content))
+        rescue JSON::ParserError
+          raise Dependabot::DependencyFileNotParseable, T.must(dependency_files.first).path
+        end
+
+        return unless contents["sdk"]
+
+        Dependabot::Dependency.new(
+          name: "dotnet-sdk",
+          version: contents["sdk"]["version"],
+          package_manager: "dotnet_sdk",
+          requirements: [{
+            file: dependency_file.name,
+            requirement: contents["sdk"]["version"],
+            groups: [],
+            source: nil
+          }],
+          metadata: {
+            allow_prerelease: contents["sdk"]["allowPrerelease"] || false,
+            roll_forward: contents["sdk"]["rollForward"] || "latestPatch"
+          }
+        )
+      end
```

Comment:
Is trying to update tools in scope too?

As an example, the ASP.NET Core team use [my GitHub Action](https://github.com/martincostello/update-dotnet-sdk) to update their .NET SDK version, and it also updates the version in `tools.dotnet` that needs to match `sdk.version`: [example](https://github.com/dotnet/aspnetcore/pull/58263/files).

Reply:
In the long run, yes I think so. But I wanted to scope this down to an MVP to start with.

Is there any documentation on this btw? I couldn't find anything in the [`global.json`][1] documentation.

[1]: https://learn.microsoft.com/en-us/dotnet/core/tools/global-json

Reply:
I don't think there's any _one-true_, or if there is I haven't found it.

It _may_ also be that this is something specific to their engineering team and I just made it work anyway. I've only seen it their repos (and stuff using their build tools) rather than generally in-the-wild.

Reply:
Okay, I'm going to mark this as resolved for now. I'll defer it for post-MVP.